### PR TITLE
fix!: header consensus

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Fix header consensus encoding

How Has This Been Tested?
---

### CODE REVIEW

File: base_layer/core/src/blocks/block_header.rs [hunk 1]
```diff
@@ -412,12 +412,12 @@ impl ConsensusEncoding for BlockHeader {
         self.height.consensus_encode(writer)?;
         copy_into_fixed_array_lossy::<_, 32>(&self.prev_hash).consensus_encode(writer)?;
         self.timestamp.consensus_encode(writer)?;
-        self.output_mr.as_slice().consensus_encode(writer)?;
-        self.witness_mr.as_slice().consensus_encode(writer)?;
+        self.output_mr.as_fixed_slice().consensus_encode(writer)?;
```
Comment (sdbondi) [inline base_layer/core/src/blocks/block_header.rs]: `FixedHash` already has a consensus encoding implementation
```suggestion
        self.output_mr.consensus_encode(writer)?;
```

**Comment by sdbondi** [comment]
You'll also need to update the tari_miner_helper consensus change test.

Bonus: change the consensus decoding code
```rust
// From this
FixedHash::from(<[u8; 32] as ConsensusDecoding>::consensus_decode(reader)?);
// to this
FixedHash::consensus_decode(reader)?;

**Comment by SWvheerden** [comment]
Thanks, 
but no, tari_miner_helper consensus change test is correct as of #4522

**Review by sdbondi — APPROVED** [review]

**Comment by sdbondi** [comment]
I suggest adding a test for header consensus encoding 

```rust
#[test]
fn consensus_encoding() {
 let consensus_constants = ConsensusConstantsBuilder::new(Network::Esmeralda)
        .build();
   let header = BlockHeader::new(consensus_constants.blockchain_version);
   check_consensus_encoding_correctness(header).unwrap();
}
```

**Comment by SWvheerden** [comment]
> I suggest adding a test for header consensus encoding
> 
> ```rust
> #[test]
> fn consensus_encoding() {
>  let consensus_constants = ConsensusConstantsBuilder::new(Network::Esmeralda)
>         .build();
>    let header = BlockHeader::new(consensus_constants.blockchain_version);
>    check_consensus_encoding_correctness(header).unwrap();
> }
> ```

yeah, will add one in the next pr

**Comment by hansieodendaal** [comment]
Thanks, these slipped me by with the latest commit in #4522

**Pull request merged**